instruments/minghe/mhs5200a.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Provides the support for the MingHe low-cost function generator.
+
+Class originally contributed by Catherine Holloway.
+"""
+
+# IMPORTS #####################################################################
+
+from __future__ import absolute_import
+from __future__ import division
+
+from builtins import range
+from enum import Enum
+
+import quantities as pq
+
+from instruments.abstract_instruments import Instrument
+from instruments.util_fns import ProxyList, assume_units
+
+# CLASSES #####################################################################
+
+
+class MHS5200(Instrument):
+    """
+    The MHS5200 is a low-cost, 2 channel function generator.
+
+    There is no user manual, but Al Williams has reverse-engineered the
+    communications protocol:
+    https://github.com/wd5gnr/mhs5200a/blob/master/MHS5200AProtocol.pdf
+    """
+    # pylint: disable=unused-variable
+    def __init__(self, filelike):
+        super(MHS5200, self).__init__(filelike)
+        self._channel_count = 2
+
+    # INNER CLASSES #
+
+    class Channel(object):
```

Comment:
I'm not sure how this would work. If there are multiple channels, would each need an open file handle? I think the system would complain about multiple instances accessing the same file.